java/java-impl/src/com/intellij/codeInsight/completion/JavaGenerateMemberCompletionContributor.java

```diff
@@ -198,15 +264,18 @@ private static LookupElement createGenerateMethodElement(PsiMethod prototype,
                                               p -> getShortParameterName(substitutor, p) + " " + p.getName(),
                                               ", ") + ")";
 
-    String overrideSignature = " @Override " + signature; // leading space to make it a middle match, under all annotation suggestions
-    LookupElementBuilder element = LookupElementBuilder.create(prototype, signature).withLookupString(methodName).
-      withLookupString(signature).withLookupString(overrideSignature).withInsertHandler(insertHandler).
+    String depriorizationPrefix = (deprioritize) ? " " : "";
```

Comment:
Why deprioritize? Why in so strange way? Making it a middle match seems a very indirect way to change sorting. Especially given that now and then people come who want some middle matches above some start matches.

Reply:
@donnerpeter I completely agree that it's a very indirect way. I reused the trick you introduced here: https://github.com/JetBrains/intellij-community/commit/ca3481dbd3f476fb03927b4a621685498d5c82d8#diff-909ebf4cda26a1ba075b0879f61dcdedR158

I wanted to preserve the old behavior that seemed to ensure that starting to type `@O` will always prefer annotation completions, e.g. maybe I wanted to type `@Overloaded` and the user shouldn't be shown suggestions for 20 different possible overrides at the top until they've typed `@Overl`

Reply:
Ouch. You're right. Apparently I had no better ideas :(